en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,179 @@
+| CONFIG_PUBLIC_KEY0           | Location of public key for keystore index 0.<br />= `{path to key0}`                                                               |
+| CONFIG_PUBLIC_KEY1           | Location of public key for keystore index 1.<br />= `{path to key1}`                                                               |
+
+::: warning
+Crypto uses a lot of flash memory, and many builds are close to their maximum capacity.
+If you run into a build error telling you that you have gone above the maximum flash memory, you will need to disable other features in the `.px4board` file you are working on, or in the `default.px4board` file.
+Be careful not to disable something you need.
+
+As an example, you could disable SIH mode by setting `CONFIG_MODULES_SIMULATION_SIMULATOR_SIH=n` which could free up enough flash memory to allow crypto to be added.
+:::
+
+## Adding Log Encryption to NuttX Config
+
+If the flight controller you are working with does not already have certain encryption settings enabled in the nuttX config, you will need to add them.
+This is done using the `kconfig` tool, which is described in [PX4 Board Configuration (Kconfig)](../hardware/porting_guide_config.md).
+
+To use Kconfig you will need to add these dependencies
+
+```sh
+sudo apt-get install libncurses-dev flex bison openssl libssl-dev dkms libelf-dev libudev-dev libpci-dev libiberty-dev autoconf
+```
+
+Now, in PX4, run the normal `make` command you would use to build the board you are targeting, but add "menuconfig" at the end of it.
+As an example:
+
+```sh
+make px4_fmu-v5_cryptotest menuconfig
+```
+
+Now go to "Crypto API".
+
+![KConfig Crypto API Main Menu Option](../../assets/hardware/kconfig-crypto-1.png)
+
+Check "Crypto API Support", "Blake2s hash algorithm", "Entropy pool and strong random number generator" and "Use interrupts to feed timing randomness to entropy pool".
+
+![KConfig Crypto Options Set](../../assets/hardware/kconfig-crypto-2.png)
+
+::: tip
+Some of these options can be tweaked if desired.
+:::
+
+After enabling encryption settings in kconfig, you may now build and test.
+
+## Download Encrypted Log Files From QGroundControl
+
+::: info
+When you download log files off of QGroundControl, both the encrypted log and the symmetric key will have the ".ulg" suffix.
+You can tell which one is the symmetric key by the file size.
+The symmetric key is normally pretty small (around 300 bytes).
+:::
+
+The process for downloading log files is the same, except you will download two separate files per log.
+Each time you fly, a log file, and a symmetric key are generated.
+The only way to tell them apart is by file size.
+
+As normal, go to **Analyze Tools -> Log Download** and select the log you wish to download.
+
+Make sure to download both the encrypted log and symmetric key that were generated.
```

Comment:
Should we perhaps note that even though they have the same filename, which sadly isn't shown, you can tell which is which based on file size?

Reply:
I have an info panel right above this that mentions it

Reply:
You are way ahead of me. Sorry!

I ran out of time on this. I will put it on my list to have another look next week.

Reply:
No worries. I'll address these comments more on the weekend / make edits

Reply:
Thank you very much. I'm back Weds on this stuff but I'll try be responsive.